snuba/web/delete_query.py

```diff
@@ -136,48 +195,88 @@ def get_new_from_clause() -> Table:
         )
 
 
-def _delete_from_table(
+def _get_attribution_info(attribution_info: Mapping[str, Any]) -> AttributionInfo:
+    info = dict(attribution_info)
+    info["app_id"] = get_app_id(attribution_info["app_id"])
+    info["referrer"] = attribution_info["referrer"]
+    info["tenant_ids"] = attribution_info["tenant_ids"]
+    return AttributionInfo(**info)
+
+
+def _execute_query(
+    query: Query,
     storage: WritableTableStorage,
     table: str,
-    conditions: Dict[str, Any],
+    cluster_name: Optional[str],
+    attribution_info: AttributionInfo,
+    query_settings: HTTPQuerySettings,
 ) -> Result:
-    cluster_name = storage.get_cluster().get_clickhouse_cluster_name()
-    on_cluster = literal(cluster_name) if cluster_name else None
-    query = Query(
-        from_clause=Table(
-            table,
-            ColumnSet([]),
-            storage_key=storage.get_storage_key(),
-            allocation_policies=storage.get_delete_allocation_policies(),
-        ),
-        condition=_construct_condition(conditions),
-        on_cluster=on_cluster,
-        is_delete=True,
-    )
-
-    columns = storage.get_schema().get_columns()
-    column_validator = ColumnValidator(columns)
-    try:
-        for col, values in conditions.items():
-            column_validator.validate(col, values)
-    except InvalidColumnType as e:
-        raise InvalidQueryException(e.message)
+    """
+    Formats and executes the delete query, taking into account
+    the delete allocation policies as well.
+    """
 
-    try:
-        _enforce_max_rows(query)
-    except NoRowsToDeleteException:
-        result: Result = {}
-        return result
+    formatted_query = format_query(query)
+    allocation_policies = _get_allocation_policies(query)
+    query_id = uuid.uuid4().hex
+    clickhouse_settings: MutableMapping[str, Any] = {"query_id": query_id}
+    result = None
+    error = None
 
-    deletion_processors = storage.get_deletion_processors()
-    # These settings aren't needed at the moment
-    dummy_query_settings = HTTPQuerySettings()
-    for deletion_procesor in deletion_processors:
-        deletion_procesor.process_query(query, dummy_query_settings)
+    stats: MutableMapping[str, Any] = {
+        "clickhouse_table": table,
+        "referrer": attribution_info.referrer,
+        "cluster_name": cluster_name or "<unknown>",
+    }
 
-    formatted_query = format_query(query)
-    # TODO error handling and the lot
-    return storage.get_cluster().get_deleter().execute(formatted_query)
+    try:
+        _apply_allocation_policies_quota(
+            query_settings,
+            attribution_info,
+            formatted_query,
+            stats,
+            allocation_policies,
```

Comment:
You're using the `_get_allocation_policies` function here which will get you the query policies. Aren't you using different policies for deleting?

Reply:
@volokluev shouldn't that function get the policies off the query? I assumed https://github.com/getsentry/snuba/pull/6181/files/b8b6fd6c3dbdb65e411abb524a272e3a78393c1e#diff-c3016e51e59d45df20ed861d6db6b02d99925816fd18b5cebe581a6c44e684dbR97 is where I set the policies, and that should be populating the delete ones

Reply:
that's a really roundabout way to do that. It's (somewhat) necessary in the execution pipeline but not here. You have the storage object. just get the delete policies from the storage

Reply:
@volokluev I still pulled out getting the allocation policies out of that function in order to be able to mock it in tests